payroll-processor-functions/src/payroll-processor-functions/Features/Payrolls/PayrollEntity.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace PayrollProcessor.Functions.Features.Payrolls
+{
+    public class PayrollEntity : TableEntity
+    {
+        public DateTimeOffset CheckDate { get; set; }
+        public Guid EmployeeId { get; set; }
+        public double GrossPayroll { get; set; }
+        public string PayrollPeriod { get; set; } = "";
+        public string EmployeeDepartment { get; set; } = "";
+
+        public static class Map
+        {
+            public static Payroll To(PayrollEntity entity)
+            {
+                return new Payroll(Guid.Parse(entity.RowKey))
+                {
+                    CheckDate = entity.CheckDate,
+                    EmployeeId = entity.EmployeeId,
+                    GrossPayroll = entity.GrossPayroll,
+                    PayrollPeriod = entity.PayrollPeriod,
+                    EmployeeDepartment = entity.EmployeeDepartment
+                };
+            }
+
+            public static PayrollEntity From(Payroll payroll)
+            {
+                return new PayrollEntity
+                {
+                    PartitionKey = payroll.CheckDate.ToString("yyyyMMdd"),
```

Comment:
Here the Partition Key is the date, which should make querying for all Payrolls within a date range possible.

It unfortunately won't make finding all Payrolls of a specific Employee possible (but that's why `EmployeePayrollEntity` exists)